### DISCUSSION TITLE
Terminate running sampler

### DISCUSSION THREAD

**Discussion by BrechtCorbeel** [Comfy-Org/ComfyUI#4719] [General]
Whenever a sampler starts I know I'm in for trouble, because either you cannot terminate a sampler unless restarting or I do not know about it why is that not possible? Currently stopping a workflow only works after the next step is finished and if a sampler goes haywire, you are screwed.

**Comment by JorgeR81** [comment]
Hopefully, we may at least have an indication that the generation is being terminated. 
https://github.com/Comfy-Org/ComfyUI_frontend/issues/695

**Comment by blepping** [comment]
@BrechtCorbeel i don't think i've ever seen a sampler go haywire but i made you a simple custom node that may make interrupts during sampling more responsive: https://gist.github.com/blepping/99aeb38d7b26a4dbbbbd5034dca8aca8

all it does is patch input/output blocks and checks to see if an interrupt occurred before actually letting the model block run. you will have a `ModelPatchFastTerminate` node once you put it in `custom_nodes` and restart ComfyUI/refresh the browser.

this approach will only work for models that ComfyUI supports model patches on, so basically only SD1x/2x and SDXL. Unfortunately it won't work for more esoteric models like Flux which can't be patched.
